components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/cipher/AESCipherTool.java

```diff
@@ -0,0 +1,173 @@
+    private static final String ALGORITHM_AES = "AES";
+    private static final String ALGORITHM_SHA_256 = "SHA-256";
+    private static final int IV_SIZE = 16;
+    private static final int SECRET_KEY_LENGTH = 16; // TODO: Make this 32 again after switching to Java 9
+
+    private final SecretKey secretKey;
+    private final SecureRandom secureRandom;
+
+    /**
+     * Create an AESCipherTool instance by passing in a user secret string.
+     *
+     * @param userSecret User secret string to encode and decode a value.
+     * @throws AESCipherToolException if any error occurs while initializing the cipher tool.
+     */
+    public AESCipherTool(String userSecret) throws AESCipherToolException {
+        this.secretKey = new SecretKeySpec(getSHA256Key(userSecret, SECRET_KEY_LENGTH), ALGORITHM_AES);
+        this.secureRandom = new SecureRandom();
+    }
+
+    /**
+     * Create an AESCipherTool instance by passing in the path of a file containing the user secret.
+     *
+     * @param userSecretFile The path to the file containing the user secret.
+     * @throws IOException            Thrown if any I/O error occurs while reading the user secret file.
+     * @throws AESCipherToolException Thrown if any error occurs while initializing the cipher tool.
+     */
+    public AESCipherTool(Path userSecretFile) throws IOException, AESCipherToolException {
+        List<String> userSecret = Files.readAllLines(userSecretFile, StandardCharsets.UTF_8);
+        Files.deleteIfExists(userSecretFile);
+
+        if (userSecret.size() > 1) {
+            throw new AESCipherToolException("Multi-line user secrets not allowed");
+        }
+
+        this.secretKey = new SecretKeySpec(getSHA256Key(userSecret.get(0), SECRET_KEY_LENGTH), ALGORITHM_AES);
+        this.secureRandom = new SecureRandom();
+    }
+
+    /**
+     * This method is used to encrypt and base64 encode a plain value.
+     *
+     * @param value Value to be encrypted.
+     * @return Encrypted value.
+     * @throws AESCipherToolException Thrown if any error occurs while initializing the cipher tool.
+     */
+    public String encrypt(String value) throws AESCipherToolException {
+        try {
+            byte[] ivByteArray = getSecureRandomBytes();
+            IvParameterSpec ivParameterSpec = new IvParameterSpec(ivByteArray);
+            Cipher encryptionCipher = Cipher.getInstance(ALGORITHM_AES_CBC_PKCS5);
+            encryptionCipher.init(Cipher.ENCRYPT_MODE, secretKey, ivParameterSpec);
+            byte[] encryptedBytes = encryptionCipher.doFinal(getBytes(value));
+            return encodeBase64(appendByteArrays(ivByteArray, encryptedBytes));
+        } catch (BadPaddingException | IllegalBlockSizeException | NoSuchPaddingException
+                | NoSuchAlgorithmException | InvalidAlgorithmParameterException | InvalidKeyException  err) {
+            log.error("Failed to encrypt value: " + value, err);
```

Comment:
Do we need to log the value in here? I think throwing the err is enough.

Reply:
Its better to log the error here, otherwise it might get missed in the log file if usage of method does not explicitly log the error when it calls the method